synapse/assets/storm/migrations/model-0.2.27.storm

```diff
@@ -0,0 +1,348 @@
+    $lib.model.migration.copyEdges($oldcpe, $newcpe)
+    $lib.model.migration.copyTags($oldcpe, $newcpe)
+    $lib.model.migration.copyData($oldcpe, $newcpe)
+
+    $refinfos = $getRefInfo($oldcpe.form())
+
+    // Iterate through the views and fix up all the references
+    for $view in $lib.view.list(deporder=$lib.true) {
+        if (not $layers.has($view.layers.0.iden)) { continue }
+
+        view.exec $view.iden {
+
+            // Fix references that point to old node to now point to new node
+            for $refinfo in $refinfos {
+                ($form, $prop, $proptype, $isarray) = $refinfo
+                if ($proptype = "ndef") {
+                    $oldvalu = $oldcpe.ndef()
+                    $newvalu = $newcpe.ndef()
+                } else {
+                    $oldvalu = $oldcpe.repr()
+                    $newvalu = $newcpe.repr()
+                }
+
+                for $n in $lib.model.migration.liftByPropValuNoNorm($form, $prop, $oldvalu) {
+                    yield $n
+
+                    if $isarray {
+
+                        // We can't just [ :$prop-=$oldvalu :$prop+=$newvalu ] because the norm() function gets called
+                        // on the array type deep down in the AST. So, instead, we have to operate on the whole array.
+
+                        $list = $lib.copy(:$prop)
+                        while $list.has($oldvalu) {
+                            $list.remove($oldvalu)
+                        }
+
+                        if (not $list.has($newvalu)) {
+                            $list.append($newvalu)
+                        }
+
+                        $lib.model.migration.setNodePropValuNoNorm($node, $prop, $list)
+
+                    } else {
+                        try {
+                            [ -:$prop :$prop=$newvalu ]
+                        } catch ReadOnlyProp as exc {
+                            // The property is readonly so we can only delete it
+                            $lib.log.warning(`{$form}:{$prop} is readonly, cannot modify. Storing node: {$node.repr()}`)
+                            $removeNode($node, $view.iden, $view.layers.0.iden)
+                            | delnode
+
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    return()
+}
+
+// This is a separate function because I was getting weird bugs from scoping
+// issues. Mainly, the $view variable being overwritten because this code was
+// scoped to the whole script.
+function main() {
+
+    for $view in $lib.view.list(deporder=$lib.true) {
+
+        if (not $layers.has($view.layers.0.iden)) { continue }
+
+        view.exec $view.iden {
+
+            $layer = $lib.layer.get()
+
+            for $oldcpe in $layer.liftByProp(it:sec:cpe) {
+                $info = $lib.model.migration.s.itSecCpe_2_170_0_internal($oldcpe)
+
+                $success = ($info.status = "success")
+
+                if $success {
```

Comment:
Any reason this isn't just `if ($info.status = "success") {` ?

Reply:
Updated.